jupyterlab_gcedetails/src/service/price_service.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ServerConnection } from '@jupyterlab/services';
+import { Option, HardwareConfiguration } from '../data/data';
+
+/** The monthly price catalog for region and sku */
+type Catalog = Map<string, Map<string, number>>;
+
+/** The average number of hours in a month, i.e., 365 / 12 * 24 */
+const HOURS_PER_MONTH = 730;
+const NO_SUSTAINED_DISCOUNT_PREFIXES = ['e2-'];
+const REGIONS: Option[] = [
```

Comment:
I'd suggest moving these out to the data constants file since they may eventually prove useful outside of prices.